src/core/SP_reader.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.20;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Attestation as SPAttestation } from "../models/Attestation.sol"; // Ignoring Offchain Attestations
+import { Attestation as VeraxAttestation, IVerax } from "src/interfaces/IVerax.sol";
+import { SP } from "src/core/SP.sol";
+/*
+    Found Information Above on Github:
+    - Verax's Attestation struct 
+    - Interface I created to call a certain function -- getAttestation()
+*/
+
+contract Reader is OwnableUpgradeable {
+    /// @custom:storage-location erc7201:ethsign.SP-Reader
+    struct ReaderStorage {
+        SP signProtocol;
+        IVerax veraxRegistry;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("ethsign.SP-Reader")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant ReaderStorageLocation = 0xb209ab9212fdbbb86ea9ba187ed37f12ac1d5679ecdf9a605a02f993a67d7400;
+
+    error Invalid();
+    error Incompatible();
+    error DoesNotExist();
+
+    function _getReaderStorage() internal pure returns (ReaderStorage storage $) {
+        assembly {
+            $.slot := ReaderStorageLocation
+        }
+    }
+
+    constructor() {
+        if (block.chainid != 31_337) {
+            _disableInitializers();
+        }
+    }
+
+    function initialize(address _spAddress, address _veraxAddress) public initializer {
+        ReaderStorage storage $ = _getReaderStorage();
+        __Ownable_init(_msgSender());
+        $.signProtocol = SP(_spAddress);
+        $.veraxRegistry = IVerax(_veraxAddress);
+    }
+
+    /**
+     * @notice Updates the address of the Verax Registry.
+     * @dev Only the owner of the contract can call this function. Reverts if the provided address is the zero address.
+     * @param _veraxAddress The new address of the Verax Registry.
+     */
+    function updateVeraxAddress(address _veraxAddress) external onlyOwner {
+        ReaderStorage storage $ = _getReaderStorage();
+        if (_veraxAddress == address(0)) revert Invalid();
+        $.veraxRegistry = IVerax(_veraxAddress);
+    }
+
+    /**
+     * @notice Updates the address of the Sign Protocol Contract.
+     * @dev Only the owner of the contract can call this function. Reverts if the provided address is the zero address.
+     * @param _spAddress The new address of the Sign Protocol Contract.
+     */
+    function updateSPAddress(address _spAddress) external onlyOwner {
+        ReaderStorage storage $ = _getReaderStorage();
+        if (_spAddress == address(0)) revert Invalid();
+        $.signProtocol = SP(_spAddress);
+    }
+
+    /**
+     * @param _id The unique id of the attestation. Takes a bytes32 as this is the datatype Verax uses
+     * @return Sign Protocol Attestation
+     */
+    function getAttestationMethod1(bytes32 _id) external view returns (SPAttestation memory) {
```

Comment:
Try to follow this style guide for function ordering: https://docs.soliditylang.org/en/latest/style-guide.html#order-of-functions

Reply:
Also I think instead of saying `Method1` and `Method2`, there is a more descriptive way to name those functions.